### PR TITLE
Add TryFrom impls for Ulid

### PR DESCRIPTION
**Pull request by j-tai**
[Analogous to Uuid](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#impl-TryFrom%3C%26str%3E-for-Uuid), this PR adds `TryFrom<&str>` ~~and `TryFrom<String>`~~ implementations for the `Ulid` type.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -356,6 +357,22 @@ impl FromStr for Ulid {
...
 
+impl TryFrom<&'_ str> for Ulid {
+    type Error = DecodeError;
+
+    fn try_from(value: &'_ str) -> Result<Self, Self::Error> {
+        Ulid::from_string(value)
+    }
+}
+
+impl TryFrom<String> for Ulid {
```
Comment (dylanhart) [inline src/lib.rs:368]: Let's remove this impl. This shouldn't ever be used as there's no upside vs calling with `&str`.

**Review by dylanhart — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (j-tai) [reply src/lib.rs:368]: Right, that's fair. Wilco

**Review by dylanhart — APPROVED** [review]

**Comment by dylanhart** [comment]
Available in `v1.1.3`

**Comment by j-tai** [comment]
> Available in `v1.1.3`

Thank you!!  :heart:

**Pull request merged**